types/tests/client.test.ts

```diff
@@ -32,26 +32,36 @@ client.getCsrfToken({ req: nextReq })
 // $ExpectType Promise<string | null>
 client.csrfToken({ req: nextReq })
 
-// $ExpectType Promise<void>
-client.signin("github", { data: "foo", redirect: false }, { login: "username" })
+// $ExpectType Promise<string | null>
+client.csrfToken({ ctx: { req: nextReq } })
+
+// $ExpectType Promise<undefined>
+client.signin("github", { callbackUrl: "foo" }, { login: "username" })
```

Comment:
We could have this extra test case:

Reply:
This wouldn't actually cover it, as `"foo"` IS a valid provider, if you have one defined in the `providers` array in `NextAuthOptions`.